docs/design/downward_api_resources_limits_requests.md

```diff
@@ -0,0 +1,262 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Downward APIs for resource limits and requests
+
+## Background
+
+Currently downward APIs (via environment variables and volume plugin) only support exposing a Pod's name, namespace, annotations, labels and its IP. This document explains the need and design to extend them to expose resources (e.g. cpu, memory, storage) limits and requests.
+
+## Motivation
+
+By exposing information about computing resource requests & limits into pods, image/application authors can be aware of the computing environment in which the pods run, and could modify container images accordingly.
```

Comment:
I would restate this a little.  How about: "It is common for software packages to configure limitations on their resource usage.  Exposing the requested and limited amounts of available resources will allow packages that do this to be configured more easily.

You can also state that docker already exposes some of this information, but we want to be runtime-agnostic in Kubernetes.